### DISCUSSION TITLE
Numbers in history?

### DISCUSSION THREAD

**Discussion by muellerto** [leo-arch/clifm#311] [General]
I tried the `h` command:

![2](https://github.com/user-attachments/assets/ac69b511-c179-4087-94e8-c7c83ff02dc2)

- What are the large numbers for? Timestamps? But I can select and execute them, it's just never successful.
- How should `edit` work? In my case it says always "line 1: command not found". I have a proper `$EDITOR` variable, so what's missing?
- The history shown there is in badly sorted: the oldest and most uninteresting command lines are on top. How should the user find a specific one from two days ago?

**Comment by leo-arch** [comment]
Yes, those numbers are timestamps. I'll take a look at it for a bit of improvement.

**Comment by leo-arch** [comment]
Update your plugins (reinstalling) and try the `h` command again:

1. Timestamps are gone.
2. Dup entries are gone.
3. Commands are sorted from newer to older.

> How should edit work? In my case it says always "line 1: command not found". I have a proper $EDITOR variable, so what's missing?

There's no `edit` subcommand for the `h` command. If you want to edit the history file run `history edit` instead (`h` is a different command; indeed, it's a plugin, while `history` is an internal command).

If you want to locate a command from a specific timestamp run `history show-time`.

**Comment by muellerto** [comment]
Hey, the history made a big step forward (in my eyes). That looks good now. Thanks a lot!

**Reply by leo-arch** [reply]
You're welcome @muellerto.
